test/functional/helpers/createCollectEndpointAsserter.js

```diff
@@ -0,0 +1,122 @@
+      // we can immediately check the number of network requests.
+      await t
+        .expect(networkLogger.edgeCollectEndpointLogs.requests.length)
+        .eql(1, "No network request to the collect endpoint was detected.");
+    }
+  };
+
+  const assertCollectNotCalled = async () => {
+    if (isSendBeaconSupported()) {
+      // The browser's sendBeacon api returns before the actual request is made. Therefore,
+      // the sendEvent promise resolves before any network request may have actually been made.
+      // In order to check that a request to the collect endpoint is never made, we would
+      // have to wait for an arbitrary period of time before checking that the number of
+      // network requests to the collect endpoint is 0. Instead of waiting for an arbitrary
+      // period of time, which is a fragile and slow solution, we'll make our best attempt
+      // at this assertion by checking that sendBeacon was never called. If supported by the
+      // browser, sendBeacon is solely used to send requests to collect, so it's a fair approximation.
+      await t
+        .expect(sendBeaconMock.getCallCount())
+        .eql(0, "A network request to the collect endpoint was detected.");
+    } else {
+      // In IE, sendBeacon isn't supported, which means we fall back to XHR. In the case
+      // of XHR, the network request will be sent before the sendEvent promise resolves, so
+      // we can immediately check the number of network requests.
+      await t
+        .expect(networkLogger.edgeCollectEndpointLogs.requests.length)
+        .eql(0, "A network request to the collect endpoint was detected.");
+    }
+  };
+
+  const assertInteractCalled = async () => {
+    await t
+      .expect(networkLogger.edgeInteractEndpointLogs.requests.length)
+      .eql(1, "No network request to the interact endpoint was detected.");
+  };
+
+  const assertInteractNotCalled = async () => {
+    await t
+      .expect(networkLogger.edgeInteractEndpointLogs.requests.length)
+      .eql(0, "A network request to the interact endpoint was detected.");
+  };
+
+  return {
+    async assertCollectCalledAndNotInteract() {
+      // The order of these matter because we need to make sure sendBeacon
+      // requests have actually been sent by the browser before checking
+      // that they weren't interact calls.
+      await assertCollectCalled();
+      await assertInteractNotCalled();
+      return networkLogger.edgeCollectEndpointLogs.requests[0];
+    },
+    async assertInteractCalledAndNotCollect() {
+      await assertCollectNotCalled();
+      await assertInteractCalled();
+      return networkLogger.edgeInteractEndpointLogs.requests[0];
+    },
+    async assertNeitherCollectNorInteractCalled() {
+      await assertCollectNotCalled();
+      await assertInteractNotCalled();
+    },
+    async reset() {
+      await networkLogger.clearLogs();
+      await sendBeaconMock.reset();
+    }
+  };
+};
```

Comment:
This file is really the meat of the refactor. It's intended to abstract the complexities of the issue away from the tests themselves. While I could have broken these methods down more to something like:

* `assertCollectCalled`
* `assertCollectNotCalled`
* `assertInteractCalled`
* `assertInteractNotCalled`

it would require the test calling the methods to understand that sometimes the order in which they are called matters (see the comment on `assertCollectCalledAndNotInteract`). To make it hard for other developers to make mistakes, I decided to take the route I have taken.